src/glyphnote.ts

```diff
@@ -36,31 +41,54 @@ export class GlyphNote extends Note {
     return this.glyph.getBoundingBox();
   }
 
-  /*
-  addToModifierContext() {
+  // Add self to modifier context. `mContext` is the `ModifierContext`
+  // to be added to.
+  addToModifierContext(mContext: ModifierContext): this {
+    this.modifierContext = mContext;
+    for (let i = 0; i < this.modifiers.length; ++i) {
+      this.modifierContext.addMember(this.modifiers[i]);
+    }
+    this.setPreFormatted(false);
     return this;
   }
-  */
 
   preFormat(): this {
+    if (!this.preFormatted && this.modifierContext) {
```

Comment:
Should an error be thrown if `this.modifierContext` is `undefined`?

Reply:
I don't know.  I followed the lead from stavenote.ts, it allows modifierContext to be null in preFormat.

Reply:
Right, this is okay.